src/containers/createRoom.js

```diff
@@ -9,35 +9,91 @@ export class createRoom extends Component {
     username: "",
   };
 
-  handleChange = (event) => {
+  handleChange(event) {
     this.setState({ [event.target.name]: event.target.value });
   };
 
-  handleSubmit = (event) => {
-    event.preventDefault();
+  getRooms() {
     const reqObj = {
       method: "POST",
       headers: {
         "Content-Type": "application/json",
         Accept: "application/json",
       },
-      body: JSON.stringify({ room: this.state }),
+      body: JSON.stringify({
+        room: this.state
+      }),
     };
-    fetch("http://localhost:3000/rooms", reqObj)
-      .then((resp) => resp.json())
-      .then((resp) => {
-        localStorage.setItem("token", resp.jwt);
-        this.props.setCreateRoom(
-          resp.user,
-          resp.room.room_name,
-          resp.room.host_id,
-          resp.room.host_name
-        );
-        this.props.history.push(`/room/${resp.room.id}`);
-      });
+
+    return fetch("http://localhost:3000/rooms", reqObj)
+      .then((resp) => resp.json());
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+
+    this.getRooms().then((response) => {
+      let { jwt, user, room} = response;
```

Comment:
When you have an object that has a collection of more/less unrelated properties, it can make a lot more sense to split them out into seperate variables.

Here, I used the destructuring syntax to take an object and destructure it into several variables based on the keys of that object. This code is the same as:

```
let jwt = response;.jwt 
let user = response.user; 
let room = response.room; 
```